pallets/avn/src/lib.rs

```diff
@@ -246,28 +275,55 @@ impl<T: Config> Pallet<T> {
 
     // TODO [TYPE: refactoring][PRI: LOW]: choose a better function name
     pub fn is_primary(
-        block_number: BlockNumberFor<T>,
+        op_type: OperationType,
         current_validator: &T::AccountId,
     ) -> Result<bool, Error<T>> {
-        let primary_validator = Self::calculate_primary_validator(block_number)?;
-        return Ok(&primary_validator == current_validator)
+        let validators = Self::validators();
+
+        // If there are no validators there's no point continuing
+        if validators.len() == 0 {
+            return Err(Error::<T>::NoValidatorsFound)
+        }
+
+        let counters = PrimaryValidator::<T>::get();
+        let primary_validator = &validators[match op_type {
+            OperationType::Ethereum => counters.ethereum,
+            OperationType::Avn => counters.avn,
+        } as usize]
```

Comment:
We shouldn't trust the counters value blindly to access in memory values, cause it can lead to a panic.

the counters value get calculated via a mod operation with `validators_len`, but that value is not constant across time.
In a system with 5 collators, should a collator leave the system, and the storage item has him as the next one, the above line will cause a panic.

Reply:
Added a check to verify if the index is not out of bounds.